master/internal/experiment.go

```diff
@@ -681,6 +681,10 @@ func (e *experiment) processOperations(
 	}
 
 	for requestID := range updatedTrials {
+		ref := ctx.Child(requestID)
+		if ref == nil {
+			panic(fmt.Sprintf("invalid request id: %v", requestID))
```

Comment:
```suggestion
			panic(fmt.Sprintf("invalid request ID: %v", requestID))
```
Hmm, I'm sort of inclined to say on general principle that this should be a log and not a panic, since it can be caused by external input. Though I guess this is no worse than the current state, and keeping it as visible as possible might be helpful. I'd say it's fine this way, just wanted to get the thoughts out.

Reply:
Yeah I did think about that too. But I thought it was better to panic here when you see it for the first time than later.